src/AndersonAcceleration.f90

```diff
@@ -85,109 +85,56 @@ MODULE AndersonAccelerationTypes
 !-------------------------------------------------------------------------------
 !> @brief Initializes the Anderson Acceleration Type with a parameter list
 !> @param pList the parameter list
-!>
 !> @param solver The anderson acceleration solver to act on
 !> @param MPIEnv The MPI environment description
-!> @param Params The aprameterlist
-!>
-!> This routine initializes the data spaces for the NOX Anderson Acceleration
+!> @param Params The anderson options parameter list
 !>
 SUBROUTINE init_AndersonAccelerationType(solver,MPIEnv,Params)
   CHARACTER(LEN=*),PARAMETER :: myName='init_AndersonAccelerationType'
   CLASS(AndersonAccelerationType),INTENT(INOUT) :: solver
   TYPE(MPI_EnvType),INTENT(IN),TARGET :: MPIEnv
   TYPE(ParamType),INTENT(IN) :: Params
 
-  TYPE(ParamType) :: validParams
-  INTEGER(SIK) :: n,nlocal,depth,start
-  REAL(SRK) :: beta
-#ifdef FUTILITY_HAVE_Trilinos_NOX
-  TYPE(ParamType) :: tmpPL
-#endif
-
-  !Check to set up required and optional param lists.
-  !IF(.NOT.AndersonType_Paramsflag) CALL AndersionType_Declare_ValidParams()
-  !Validate against the reqParams and OptParams
-  validParams=Params
-  !CALL validParams%validate(AndersonType_reqParams)
+  INTEGER(SIK) :: i,M
 
+  !Check to ensure parallel environment is initialized
   IF(.NOT. MPIEnv%isInit()) THEN
     CALL eAndersonAccelerationType%raiseError('Incorrect input to '// &
         modName//'::'//myName//' - MPI Environment is not initialized!')
   ELSE
     solver%MPIparallelEnv => MPIEnv
```

Comment:
You never use this.  Do you need to?

Reply:
Don't need it, like Aaron pointed out we can use the Futility parallel environment.

Reply:
strictly speaking, the vector class will hold its own parallel env.  When you use the BLAS routines, those should be handled under the hood.